### PR TITLE
Release notes `0.27.0`

### PR DESCRIPTION
**Pull request by isaacdevlugt**

### CODE REVIEW

File: doc/releases/changelog-0.27.0.md [hunk 1]
```diff
@@ -4,57 +4,226 @@
...
+* [(#3195)](https://github.com/PennyLaneAI/pennylane/pull/3195)
+* [(#3194)](https://github.com/PennyLaneAI/pennylane/pull/3194)
+* [(#3184)](https://github.com/PennyLaneAI/pennylane/pull/3184)
+* [(#3249)](https://github.com/PennyLaneAI/pennylane/pull/3249)
+* [(#3267)](https://github.com/PennyLaneAI/pennylane/pull/3267)
+* [(#3234)](https://github.com/PennyLaneAI/pennylane/pull/3234)
+* [(#3233)](https://github.com/PennyLaneAI/pennylane/pull/3233)
+* [(#3232)](https://github.com/PennyLaneAI/pennylane/pull/3232)
+* [(#3223)](https://github.com/PennyLaneAI/pennylane/pull/3223)
+* [(#3222)](https://github.com/PennyLaneAI/pennylane/pull/3222)
```
Comment (isaacdevlugt) [inline doc/releases/changelog-0.27.0.md:38]: These were marked with a *** in [this spreadsheet](https://docs.google.com/spreadsheets/d/1tYaVRHDqsrzSFZviJ0VRPDtR4BiOfrYhC8_LjJYK2TU/edit#gid=1295754102)

**Comment by Jaybsoni** [comment]
added missing contributors

File: doc/releases/changelog-0.27.0.md [hunk 2]
```diff
@@ -592,7 +740,7 @@
 * Added a "Deprecations" page to the developer documentation.
   [(#3093)](https://github.com/PennyLaneAI/pennylane/pull/3093)
 
-* The example of the `FlipSign` template has been updated.
+* The example of the `qml.FlipSign` template has been updated.
   [(#3219)](https://github.com/PennyLaneAI/pennylane/pull/3219)
 
 <h3>Bug fixes</h3>
```
Comment (Jaybsoni) [inline doc/releases/changelog-0.27.0.md:746]: lines 789 and 792 have the same GitHub link, can probably make them just one bullet point

File: doc/releases/changelog-0.27.0.md [hunk 3]
```diff
@@ -236,139 +397,133 @@
...
+  For more details, please
+  [refer to the documentation of `qml.enable_return()`](https://docs.pennylane.ai/en/stable/code/api/pennylane.enable_return.html?highlight=enable_return#pennylane.enable_return).
+
 <h3>Improvements</h3>
 
+* The `Exp` class decomposes into a `PauliRot` class if the coefficient is imaginary and the base operator is a Pauli Word.
+  [(#3249)](https://github.com/PennyLaneAI/pennylane/pull/3249)
+
 * Added a `samples_computational_basis` attribute to the `MeasurementProcess` and `QuantumScript` classes to track if computational basis samples are being generated when `qml.sample` or `qml.counts` are called without specifying an observable.
   [(#3207)](https://github.com/PennyLaneAI/pennylane/pull/3207)
```
Comment (Jaybsoni) [inline doc/releases/changelog-0.27.0.md:528]: This one is listed twice, once here and once in bug fixes

Comment (Jaybsoni) [reply doc/releases/changelog-0.27.0.md:528]: The PR with tag #3181 is also listed twice, once in improvements and once in bug fixes

File: doc/releases/changelog-0.27.0.md [hunk 4]
```diff
@@ -4,145 +4,224 @@
...
+  - Once a dataset is loaded, its properties can be accessed as follows:
 
-  The control wires and values that define the operation are defined analogously to the qubit operation.
+    ```pycon
+    >>> dev = qml.device('default.qubit',wires=H2data.hamiltonian.wires)
+    >>> @qml.qnode(dev)
+    ... def circuit():
+    ...     return qml.expval(H2data.hamiltonian)
+    >>> print(circuit())
+    2.173913043478261
```
Comment (DSGuala) [inline doc/releases/changelog-0.27.0.md:76]:
```suggestion
    >>> dev = qml.device("default.qubit",wires=4)
    >>> @qml.qnode(dev)
    ... def circuit():
    ...     qml.BasisState(H2data.hf_state, wires = [0, 1, 2, 3])
    ...     for op in H2data.vqe_gates:
    ...          qml.apply(op)
    ...     return qml.expval(H2data.hamiltonian)
    >>> print(circuit())
    -1.0791430411076344
```

File: doc/releases/changelog-0.27.0.md (hunk 2 above)
Comment (isaacdevlugt) [reply doc/releases/changelog-0.27.0.md:742]: Sorry a little confused here, there's <700 lines in the changelog 🥴

Comment (isaacdevlugt) [reply doc/releases/changelog-0.27.0.md:742]: Thanks!

File: doc/releases/changelog-0.27.0.md [hunk 5]
```diff
@@ -4,145 +4,227 @@
...
+    ```
+
+    The available attributes can be found using `qml.data.list_attributes()`:
+
+  - To select data interactively, we can use `qml.data.load_interactive`:
 
   ```pycon
-  >>> circuit()
-  -0.42264973081037416
+  >>> qml.data.load_interactive()
```
Comment (obliviateandsurrender) [inline doc/releases/changelog-0.27.0.md:41]: Might want to reduce the indentation of the print below by one step to not have the horizontal scroll.

File: doc/releases/changelog-0.27.0.md [hunk 6]
```diff
@@ -4,145 +4,227 @@
...
+      data_name="qchem", molname="H2", basis="STO-3G", bondlength=1.1,
+      attributes=["molecule", "hamiltonian"]
+    )[0]
+    >>> H2_hamiltonian.hamiltonian
+    <Hamiltonian: terms=15, wires=[0, 1, 2, 3]>
+    ```
+
+    The available attributes can be found using `qml.data.list_attributes()`:
+
+  - To select data interactively, we can use `qml.data.load_interactive`:
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:38]:
```suggestion
  - To select data interactively, we can use `qml.data.load_interactive()`:
```

File: doc/releases/changelog-0.27.0.md [hunk 7]
```diff
@@ -4,145 +4,227 @@
...
+        Force download files? (Default is no) [y/N]: N
+        Folder to download to? (Default is pwd, will download to /datasets subdirectory): /Users/jovyan/Downloads
+
+        Please confirm your choices:
+        dataset: qspin/Ising/open/rectangular/4x4
+        attributes: ['parameters', 'ground_states']
+        force: False
+        dest folder: /Users/jovyan/Downloads/datasets
+        Would you like to continue? (Default is yes) [Y/n]:
+        <Dataset = description: qspin/Ising/open/rectangular/4x4, attributes: ['parameters', 'ground_states']>
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:65]: Could we deindent this so that it's at the same depth as the first > in line 41? This is how it appears to me on a console.
```suggestion
  Please select a data name:
      1) qspin
      2) qchem
  Choice [1-2]: 1
  Please select a sysname:
      ...
  Please select a periodicity:
      ...
  Please select a lattice:
      ...
  Please select a layout:
      ...
  Please select attributes:
      ...
  Force download files? (Default is no) [y/N]: N
  Folder to download to? (Default is pwd, will download to /datasets subdirectory):

  Please confirm your choices:
  dataset: qspin/Ising/open/rectangular/4x4
  attributes: ['parameters', 'ground_states']
  force: False
  dest folder: datasets
  Would you like to continue? (Default is yes) [Y/n]:
  <Dataset = description: qspin/Ising/open/rectangular/4x4, attributes: ['parameters', 'ground_states']>
```

File: doc/releases/changelog-0.27.0.md [hunk 8]
```diff
@@ -4,145 +4,227 @@
...
-      qml.TShift(wires=0)
-      qml.TAdd(wires=[0, 1])
-      qml.ControlledQutritUnitary(U, control_wires=[0, 1], control_values='12', wires=2)
-      return qml.state()
+  ```pycon
+  >>> example_hamiltonian = qml.Hamiltonian(coeffs=[1,0.5], observables=[qml.PauliZ(wires=0),qml.PauliX(wires=1)])
+  >>> example_energies, _ = np.linalg.eigh(qml.matrix(example_hamiltonian))
+  >>> example_dataset = qml.data.Dataset(
+    data_name = 'Example', hamiltonian=example_hamiltonian, energies=example_energies
+  )
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:89]:
```suggestion
... data_name = 'Example', hamiltonian=example_hamiltonian, energies=example_energies
... )
```

File: doc/releases/changelog-0.27.0.md [hunk 9]
```diff
@@ -4,145 +4,227 @@
...
+  >>> example_dataset.data_name
+  'Example'
+  >>> example_dataset.hamiltonian
+      (0.5) [X1]
+  + (1) [Z0]
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:94]:
```suggestion
    (0.5) [X1]
  + (1) [Z0]
```

File: doc/releases/changelog-0.27.0.md [hunk 10]
```diff
@@ -4,145 +4,227 @@
...
-  >>> unitaries
-  [tensor([[-1.00000000e+00, -5.46483514e-13],
-         [ 5.46483514e-13, -1.00000000e+00]], requires_grad=True),
-  tensor([[-1.00000000e+00,  3.17585063e-14],
-          [-3.17585063e-14, -1.00000000e+00]], requires_grad=True),
-  tensor([[-0.70710678, -0.70710678],
-          [-0.70710678,  0.70710678]], requires_grad=True),
-  tensor([[ 2.58789009e-11,  1.00000000e+00],
-          [-1.00000000e+00,  2.58789009e-11]], requires_grad=True)]
+  The `qml.AdaptiveOptimizer` optimizer takes an initial circuit and a collection of operators as input and adds a selected gate to the circuit at each optimization step. The process of growing the circuit can be repeated until the circuit gradients converge to zero within a given threshold. The adaptive optimizer can be used to implement algorithms such as ADAPT-VQE as shown in the following example.
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:119]:
```suggestion
  The `qml.AdaptiveOptimizer` takes an initial circuit and a collection of operators as input and adds a selected gate to the circuit at each optimization step. The process of growing the circuit can be repeated until the circuit gradients converge to zero within a given threshold. The adaptive optimizer can be used to implement algorithms such as ADAPT-VQE as shown in the following example.
```

File: doc/releases/changelog-0.27.0.md [hunk 11]
```diff
@@ -4,145 +4,227 @@
...
-[(#3113)](https://github.com/PennyLaneAI/pennylane/pull/3113)
+<h4>Automatic interface detection 🧩</h4>
 
-  Given a basis state :math:`\vert n \rangle`, where :math:`n` is a positive integer, and a fixed positive integer :math:`L`, the `IntegerComparator` operator flips a target qubit if :math:`n \geq L`. Alternatively, the flipping condition can be :math:`n < L` as demonstrated below:
+* QNodes now accept an `auto` interface argument which automatically detects the machine learning library to use.
+  [(#3132)](https://github.com/PennyLaneAI/pennylane/pull/3132)
 
   ```python
   dev = qml.device("default.qubit", wires=2)
+  @qml.qnode(dev, interface="auto")
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:205]:
```suggestion
  from pennylane import numpy as np
  import torch
  import tensorflow as tf
  from jax import numpy as jnp

  dev = qml.device("default.qubit", wires=2)
  @qml.qnode(dev, interface="auto")
```

File: doc/releases/changelog-0.27.0.md [hunk 12]
```diff
@@ -174,32 +256,112 @@
 
   Note that this change depends on `jax.pure_callback`, which requires `jax>=0.3.17`.
 
-* The `QNode` class now accepts an `auto` interface, which automatically detects the interface of the given input.
-  [(#3132)](https://github.com/PennyLaneAI/pennylane/pull/3132)
+<h4>The `qml.pauli` module 👀</h4>
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:259]:
```suggestion
<h4>Construct Pauli words and sentences 🔤</h4>
```

File: doc/releases/changelog-0.27.0.md [hunk 13]
```diff
@@ -174,32 +256,112 @@
...
-      qml.RY(weight[1], wires=1)
-      return qml.expval(qml.PauliZ(0))
+* We've reorganized and grouped everything in PennyLane responsible for manipulating Pauli operators into a `pauli` module. The `grouping` module has been deprecated as a result, and logic was moved from `pennylane/grouping` to `pennylane/pauli/grouping`.
+  [(#3179)](https://github.com/PennyLaneAI/pennylane/pull/3179)
 
-  interface_tensors = [[0, 1], np.array([0, 1]), torch.Tensor([0, 1]), tf.Variable([0, 1], dtype=float), jnp.array([0, 1])]
-  for tensor in interface_tensors:
-      res = circuit(weight=tensor)
-      print(f"Result value: {res:.2f}; Result type: {type(res)}")
+* `qml.pauli.PauliWord` and `qml.pauli.PauliSentence` can be used to represent tensor products and linear combinations of Pauli operators, respectively. These provide a more performant method to compute sums and products of Pauli operators.
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:264]: Maybe worth swapping the order of these two points? As a user I wasn't that excited by the reorganization point.

File: doc/releases/changelog-0.27.0.md [hunk 14]
```diff
@@ -174,32 +256,112 @@
...
+    -1.44 * I
+    + (-0.6+0j) * Z(0)
+    >>> (ps1 + ps2).to_mat(wire_order=[0,1])
+    array([[ 0. +0.j,  0. +0.j,  0.5+0.j,  0. +0.j],
+          [ 0. +0.j,  0. +0.j,  0. +0.j, -0.5+0.j],
+          [-0.5+0.j,  0. +0.j,  0. +0.j,  0. +0.j],
+          [ 0. +0.j,  0.5+0.j,  0. +0.j,  0. +0.j]])
+    ```
+
+<h4>New basis rotation and tapering features in `qml.qchem` 🤓</h4>
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:304]: Do the backticks show up ok on the website?

File: doc/releases/changelog-0.27.0.md [hunk 15]
```diff
@@ -174,32 +256,112 @@
...
+    >>> geometry =  np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.4589]])
+    >>> mol = qml.qchem.Molecule(symbols, geometry, charge=1)
+    >>> H, n_qubits = qml.qchem.molecular_hamiltonian(symbols, geometry)
+    >>> generators = qml.qchem.symmetry_generators(H)
+    >>> paulixops = qml.qchem.paulix_ops(generators, n_qubits)
+    >>> paulix_sector = qml.qchem.optimal_sector(H, generators, mol.n_electrons)
+    >>> tap_op = qml.qchem.taper_operation(qml.SingleExcitation, generators, paulixops,
+    ...                paulix_sector, wire_order=H.wires, op_wires=[0, 2])
+    >>> tap_op(3.14159)
+    [Exp(1.570795j, 'PauliY', wires=[0])]
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:341]:
```suggestion
    [Exp(1.5707949999999993j PauliY)]
```
This is how it showed on my laptop :thinking:

File: doc/releases/changelog-0.27.0.md [hunk 16]
```diff
@@ -174,32 +256,112 @@
...
+  ```pycon
+    >>> dev = qml.device('default.qubit', wires=[0, 1])
+    >>> @qml.qnode(dev)
+    ... def circuit(params):
+    ...     tap_op(params[0])
+    ...     return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+    >>> drawer = qml.draw(circuit, show_all_wires=True)
+    >>> print(drawer(params=[3.14159]))
+        0: ─Exp(1.570795j PauliY)─┤ ╭<Z@Z>
+        1: ───────────────────────┤ ╰<Z@Z>
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:355]:
```suggestion
    0: ──Exp(0.00+1.57j Y)─┤ ╭<Z@Z>
    1: ────────────────────┤ ╰<Z@Z>
```

**Review by trbromley — CHANGES_REQUESTED** [review]
Thanks @isaacdevlugt!

File: doc/releases/changelog-0.27.0.md (hunk 14 above)
Comment (isaacdevlugt) [reply doc/releases/changelog-0.27.0.md:390]: Hmm best take them out I think

File: doc/releases/changelog-0.27.0.md (hunk 13 above)
Comment (isaacdevlugt) [reply doc/releases/changelog-0.27.0.md]: I think I want to keep the order as it is. If a user sees `qml.pauli` first, they might be a little confused

File: doc/releases/changelog-0.27.0.md [hunk 17]
```diff
@@ -174,32 +264,190 @@
...
+                [ 0. ,  0. ]]], dtype=float32)),
+  (DeviceArray([[ 0.07677898,  0.0563341 ],
+                [ 0.07238522, -1.830669  ]], dtype=float32),
+  DeviceArray([[[-4.9707499e-01,  2.9999996e-04],
+                [-6.2500127e-04,  1.2500001e-04]],
+                [[ 4.9707499e-01, -2.9999996e-04],
+                [ 6.2500127e-04, -1.2500001e-04]]], dtype=float32)))
   ```
 
+  For more details, please [refer to the documentation of `qml.enable_return()`](https://docs.pennylane.ai/en/stable/code/api/pennylane.enable_return.html?highlight=enable_return#pennylane.enable_return).
```
Comment (josh146) [inline doc/releases/changelog-0.27.0.md:388]:
```suggestion
  For more details, please [refer to the documentation](https://docs.pennylane.ai/en/stable/code/api/pennylane.enable_return.html?highlight=enable_return#pennylane.enable_return).
```

image

File: doc/releases/changelog-0.27.0.md [hunk 18]
```diff
@@ -174,32 +264,190 @@
...
+  - Gradient transforms
+    + `qml.gradients.param_shift`
+    + `qml.gradients.finite_diff`
+    + `qml.gradients.hessian_transform`
+    + `qml.gradients.param_shift_hessian`
+
+  - Interfaces 
+    + Autograd
+    + TensorFlow
+    + JAX (without jitting)
```
Comment (josh146) [inline doc/releases/changelog-0.27.0.md:352]:
```suggestion
  - Interfaces (Autograd, TensorFlow, and JAX, although without JIT)
```

File: doc/releases/changelog-0.27.0.md [hunk 19]
```diff
@@ -174,32 +264,190 @@
...
+
+  With this release, we're continuing to elevate this feature by adding support for:
+  
+  - `qml.execute`
+  - Jacobian vector product (JVP) computation
+  - Gradient transforms
+    + `qml.gradients.param_shift`
+    + `qml.gradients.finite_diff`
+    + `qml.gradients.hessian_transform`
+    + `qml.gradients.param_shift_hessian`
```
Comment (josh146) [inline doc/releases/changelog-0.27.0.md:347]:
```suggestion
  - Execution (`qml.execute`)
  - Jacobian vector product (JVP) computation
  - Gradient transforms (`qml.gradients.param_shift`, `qml.gradients.finite_diff`, qml.gradients.hessian_transform`, `qml.gradients.param_shift_hessian`).
```

@isaacdevlugt a _very_ minor suggestion, feel free not to accept.

File: doc/releases/changelog-0.27.0.md [hunk 20]
```diff
@@ -174,32 +264,183 @@
...
+  [(#3222)](https://github.com/PennyLaneAI/pennylane/pull/3222)
+  [(#3315)](https://github.com/PennyLaneAI/pennylane/pull/3315)
+
+  In v0.25, we introduced `qml.enable_return()`, which separates measurements into their own tensors. The motivation of this change is the deprecation of ragged `ndarray` creation in NumPy.
+
+  With this release, we're continuing to elevate this feature by adding support for:
+  
+  - Execution (`qml.execute`)
+  - Jacobian vector product (JVP) computation
+  - Gradient transforms (`qml.gradients.param_shift`, `qml.gradients.finite_diff`, qml.gradients.hessian_transform`, `qml.gradients.param_shift_hessian`).
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:343]:
```suggestion
  - Gradient transforms (`qml.gradients.param_shift`, `qml.gradients.finite_diff`, `qml.gradients.hessian_transform`, `qml.gradients.param_shift_hessian`).
```

File: doc/releases/changelog-0.27.0.md [hunk 21]
```diff
@@ -174,32 +264,183 @@
...
+    >>> geometry =  np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.4589]])
+    >>> mol = qml.qchem.Molecule(symbols, geometry, charge=1)
+    >>> H, n_qubits = qml.qchem.molecular_hamiltonian(symbols, geometry)
+    >>> generators = qml.qchem.symmetry_generators(H)
+    >>> paulixops = qml.qchem.paulix_ops(generators, n_qubits)
+    >>> paulix_sector = qml.qchem.optimal_sector(H, generators, mol.n_electrons)
+    >>> tap_op = qml.qchem.taper_operation(qml.SingleExcitation, generators, paulixops,
+    ...                paulix_sector, wire_order=H.wires, op_wires=[0, 2])
+    >>> tap_op(3.14159)
+    [Exp(1.5707949999999993j PauliY)]
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:420]:
```suggestion
  >>> symbols = ['He', 'H']
  >>> geometry =  np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.4589]])
  >>> mol = qml.qchem.Molecule(symbols, geometry, charge=1)
  >>> H, n_qubits = qml.qchem.molecular_hamiltonian(symbols, geometry)
  >>> generators = qml.qchem.symmetry_generators(H)
  >>> paulixops = qml.qchem.paulix_ops(generators, n_qubits)
  >>> paulix_sector = qml.qchem.optimal_sector(H, generators, mol.n_electrons)
  >>> tap_op = qml.qchem.taper_operation(qml.SingleExcitation, generators, paulixops,
  ...                paulix_sector, wire_order=H.wires, op_wires=[0, 2])
  >>> tap_op(3.14159)
  [Exp(1.5707949999999993j PauliY)]
```

File: doc/releases/changelog-0.27.0.md [hunk 22]
```diff
@@ -174,32 +264,183 @@
...
+  ```pycon
+    >>> dev = qml.device('default.qubit', wires=[0, 1])
+    >>> @qml.qnode(dev)
+    ... def circuit(params):
+    ...     tap_op(params[0])
+    ...     return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+    >>> drawer = qml.draw(circuit, show_all_wires=True)
+    >>> print(drawer(params=[3.14159]))
+    0: ──Exp(0.00+1.57j Y)─┤ ╭<Z@Z>
+    1: ────────────────────┤ ╰<Z@Z>
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:434]:
```suggestion
  >>> dev = qml.device('default.qubit', wires=[0, 1])
  >>> @qml.qnode(dev)
  ... def circuit(params):
  ...     tap_op(params[0])
  ...     return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
  >>> drawer = qml.draw(circuit, show_all_wires=True)
  >>> print(drawer(params=[3.14159]))
  0: ──Exp(0.00+1.57j Y)─┤ ╭<Z@Z>
  1: ────────────────────┤ ╰<Z@Z>
```

File: doc/releases/changelog-0.27.0.md [hunk 23]
```diff
@@ -4,145 +4,235 @@
...
+        ...
+    Please select a lattice:
+        ...
+    Please select a layout:
+        ...
+    Please select attributes:
+        ...
+    Force download files? (Default is no) [y/N]: N
+    Folder to download to? (Default is pwd, will download to /datasets subdirectory):
+
```
Comment (Jaybsoni) [inline doc/releases/changelog-0.27.0.md:58]: Extra newline here?

**Review by Jaybsoni — APPROVED** [review]
Just one small comment! Everything else looks great to me! 💯 🎉

File: doc/releases/changelog-0.27.0.md [hunk 24]
```diff
@@ -236,143 +480,84 @@
...
+  >>> circuit(U)
+  tensor([0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j,
+        0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 1.+0.j, 0.+0.j, 0.+0.j, 0.+0.j,
+        0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j,
+        0.+0.j, 0.+0.j, 0.+0.j], requires_grad=True)
   ```
 
 <h3>Improvements</h3>
 
 * Adds a Python 3.11 classification to the PennyLane package.
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:557]:
```suggestion
* PennyLane now supports Python 3.11!
```

File: doc/releases/changelog-0.27.0.md [hunk 25]
```diff
@@ -673,16 +827,20 @@ This release contains contributions from (in alphabetical order):
 Kamal Mohamed Ali,
 Guillermo Alonso-Linaje,
 Juan Miguel Arrazola,
-Albert Mitjans Coma,
 Utkarsh Azad,
+Tom Bromley,
```
Comment (trbromley) [inline doc/releases/changelog-0.27.0.md:831]:
```suggestion
Thomas Bromley, 
```
Thanks!

**Review by trbromley — APPROVED** [review]
Thanks @isaacdevlugt! I've left some suggestions, but otherwise good to merge!

File: doc/releases/changelog-0.27.0.md [hunk 26]
```diff
@@ -4,145 +4,235 @@
...
+  - Datasets available to be downloaded can be listed with `qml.data.list_datasets()`.
 
-  @qml.qnode(dev)
-  def circuit():
-      qml.TClock(wires=0)
-      qml.TShift(wires=1)
-      qml.TAdd(wires=[0, 1])
-      return qml.expval(qml.GellMann(wires=0, index=8) + qml.GellMann(wires=1, index=3))
-  ```
+  - To download or load only specific properties of a dataset, we can specify the desired properties in `qml.data.load` with the `attributes` keyword argument:
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply doc/releases/changelog-0.27.0.md:25]: to be honest, slight implementation secret but `attributes` only affects what's _downloaded_. The way `load` works is:
1. Determine all descriptions (combinations of parameters other than `attributes`) the user is requesting
2. Download one file per-attribute, per-description (the default "full" attribute gets a "_full.dat" file) to a folder following a naming format (eg. `datasets/qchem/H2/STO-3G/1.1/`)
3. Create a `Dataset`, and give it that pathname. It doesn't even take the list of attributes! It just scans the folder and gets any attributes there. This means that if say, yesterday you did `qml.data.load("qchem", molname="H2", basis="STO-3G", bondlength=1.1)` (which downloads the `_full.dat` file), but today you called that with `attributes=["molecule"]`, then did `my_dataset.list_attributes()`, you'd find that they'd all be there!

Ultimately, I left that implementation because we have lazy-loading implemented. While `my_dataset.hamiltonian` may be massive, it isn't actually saved to memory at all. This can be changed in the future if we don't like it, but that's the state of things today.

tl;dr specifying `attributes` changes what is _downloaded_, and while not direct, that does have some effect on what is then loaded to memory.
